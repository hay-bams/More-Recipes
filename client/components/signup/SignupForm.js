import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @class SignupForm
 */
class SignupForm extends React.Component {
    render() {
        return(
            <div className = "container-fluid main-register-container">
            <div className = "row">
              <div className = "col-sm-8 col-md-5 mx-auto">
                <div className = "card  mt-5">
                  <div className = "card-header">
                    Register
                  </div>
      
                  <div className = "card-body">
                    <form>
                       <div className = "form-group">
                        <label htmlFor = "first_name">First Name</label>
                        <input type="text" name="first_name" className = "form-control" placeholder="Enter First name" />
                      </div>
      
                       <div className = "form-group">
                        <label htmlFor = "last_name">Last Name</label>
                        <input type="text" name="last_name" className = "form-control" placeholder="Enter Last name" />
                      </div>
      
                      <div className = "form-group">
                        <label htmlFor = "email">Email</label>
                        <input type="email" name="email" className = "form-control" placeholder="Enter Email" />
                      </div>
      
                      <div className = "form-group">
                        <label htmlFor = "password">Password</label>
                        <input type="password" name="password" className = "form-control" placeholder="Enter Password" />
                      </div>
      
                       <div className = "form-group">
                        <label htmlFor = "password">Confirm Password</label>
                        <input type="password" name="password" className = "form-control" placeholder="Confirm Password" />
                      </div>
      
                      <div className = "form-group">
                        <input type="submit" value = "Register" className = "btn btn-success btn-block" />
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        )
    }
}

export default SignupForm;