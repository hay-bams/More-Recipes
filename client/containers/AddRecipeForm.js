import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'; 
import { Link } from 'react-router-dom';
import { addRecipe } from '../actions/actions.js';
import { clearError } from '../actions/actions.js';

/**
 * @class AddRecipeForm
 */
class AddRecipeForm extends React.Component {
  constructor(props) {
    super(props);
    this.addRecipe =  this.addRecipe.bind(this);
    this.recipeNum = this.props.recipes.length;
  }

  componentWillMount() {
     this.props.clearError()
  }

  addRecipe(event) {
    event.preventDefault();
    const recipe = {
      title: this.title.value,
      ingredients: this.ingredients.value,
      instructions: this.instructions.value,
      image: 'imageURl'
    }
    this.props.addRecipe(recipe);
  } 

  render() {
    return (
      <div className = "main-userboard-body">
        <div className = "container-fluid">
          <ol className="breadcrumb">
            <li className="breadcrumb-item">
              <a href="#">Dashboard</a>
            </li>
            <li className="breadcrumb-item active">Add Recipe</li>
          </ol>
        </div>
  
        <div className = "container">
          <div className = "row">
            <div className = "col-sm-12 col-md-8 col-lg-6">
           {this.recipeNum = this.props.recipes.length - this.recipeNum === 1 ? 
            this.props.clearError() && <span></span> : ""} 
              <span>{this.props.message.errors ? this.props.message.errors.response.data.message: ""}</span>
                <form onSubmit = {this.addRecipe}>
                <div className = "form-group">
                  <label htmlFor = "food">Recipe Name</label>
                  <input ref = {(input) => this.title = input} type="text" name="title" 
                  className = "form-control" placeholder="Enter Recipe name" />
                </div>
  
                <div className = "form-group">
                  <input type="file" />
                </div>
  
                  <div className = "form-group">
                  <label htmlFor = "ingredient">Ingredients</label>
                  <input  ref = {(input) => this.ingredients = input}ype="text" name="ingredient" 
                  className = "form-control" placeholder="Enter Ingredients" />
                </div>
  
                <div className = "form-group">
                  <label htmlFor = "instruction">Instructions</label>
                  <textarea ref = {(input) => this.instructions = input} className = "form-control" 
                  placeholder="Enter Instructions" name = "instruction"></textarea>
                </div>
  
                <div className = "form-group">
                  <input type="submit" className = "btn btn-success" value = "Add Recipe" />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
  
    )
  }
}

const mapStateToProps = state => {
  return {
    recipes: state.recipes,
    message: state.errors
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({addRecipe, clearError}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(AddRecipeForm);