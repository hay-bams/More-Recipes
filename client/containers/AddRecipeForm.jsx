import React from 'react';
import axios from 'axios';
import Dropzone from 'react-dropzone';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Loader from 'react-loader';
import { addRecipe } from '../actions/recipe';
import Authenticate from '../auth/auth';

const dropZoneStyles = {
  border: 'none',
  cursor: 'pointer'
};

/**
 * @class AddRecipeForm
 *
 */
export class AddRecipeForm extends React.Component {
  /**
   * @param {obj} props
   * @returns {void} constructor
   */
  constructor(props) {
    super(props);
    this.addRecipe = this.addRecipe.bind(this);
    this.uploadImageToCloudinary = this.uploadImageToCloudinary.bind(this);
    this.handleDrop = this.handleDrop.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      imageURI: null,
      image: null,
      loaded: true,
      uploadError: null,
      ingredientError: '',
      titleError: '',
      instructionError: '',
      addingRecipe: false
    };
  }

  /**
   * @param {obj} event
   * @returns {void} onChange
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   *
   * @param {obj} file
   * @return {void} handleDrop
   */
  handleDrop(file) {
    this.setState({ image: file[0] });
  }

  /**
   *
   * @param {obj} event
   * @return {obj} uploadImageToCloudinary
   */
  async uploadImageToCloudinary() {
    try {
      const imageFile = this.state.image;
      const CLOUDINARY_URL = 'https://api.cloudinary.com/v1_1/dsj9ygnq2/upload';
      const CLOUDINARY_UPLOAD_PRESET = 'xmklgrkm';
      const formData = new FormData();
      formData.append('file', imageFile);
      formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
      const response = await axios({
        url: CLOUDINARY_URL,
        method: 'POST',
        headers: { 'X-Requested-With': 'XMLHttpRequest' },
        data: formData
      });
      return Promise.resolve(response.data.secure_url);
    } catch (err) {
      return Promise.reject(err);
    }
  }

  /**
   * @param {obj} event
   * @returns {void} addRecipe
   */
  async addRecipe(event) {
    try {
      event.preventDefault();
      let recipe = {
        title: this.state.title,
        ingredients: this.state.ingredients,
        instructions: this.state.instructions,
      };

      const errors = Authenticate.validateAddRecipe(recipe);

      if (errors.title !== '' || errors.ingredients !== '' ||
     errors.instructions) {
        return this.setState({
          ingredientError: errors.ingredients,
          titleError: errors.title,
          instructionError: errors.instructions
        });
      }
      this.setState({ loaded: false, addingRecipe: true });
      const imageURI = await this.uploadImageToCloudinary();

      recipe = {
        title: this.state.title,
        ingredients: this.state.ingredients,
        instructions: this.state.instructions,
        image: imageURI
      };

      await this.props.addRecipe(recipe);

      this.props.history.push('/view_recipes');
    } catch (err) {
      if (err.response.data.error.message) {
        this.setState({
          loaded: true,
          uploadError: 'image is required',
          addingRecipe: false,
          ingredientError: '',
          titleError: '',
          instructionError: ''
        });
      }
    }
  }

  /**
   * @returns {obj} render
   */
  render() {
    let imageSource = null;
    if (this.state.image) {
      imageSource = this.state.image.preview;
    }
    const { uploadError } = this.state;
    return (
      <div className="main-userboard-body add-recipe-body">
        <div className="container">
          <div className="row">
            <div className="mx-auto col-sm-12 col-md-8 col-lg-8 col-xs mt-5">
              <h2 className="text-center"> Add Recipe</h2>
              <form onSubmit={this.addRecipe}>
                <Dropzone
                  onDrop={this.handleDrop}
                  multiple
                  accept="image/*"
                  style={dropZoneStyles}
                >

                  {this.state.image ?
                    <img className="card-img-top" style={{ height: '450px' }} alt="" src={imageSource} /> :
                    <div className="upload-recipe-img">
                      <div className="row justify-content-center">
                        <div className="col-12">
                          <p className="text-center" id="addImage">
                            <span className="h2"><i className="ion ion-camera" /></span>
                            <br />
                         Click to upload image
                          </p>
                        </div>
                      </div>
                    </div>
                 }
                </Dropzone>
                <div>
                  { uploadError &&
                  <span className="help-block error text-danger">
                    {uploadError}
                  </span>
                    }
                </div>

                <img src={`${this.state.imageURI}`} id="imagePreview" alt="" />
                <div className="form-group">
                  <label htmlFor="food">Recipe Name</label>
                  <input
                    onChange={this.onChange}
                    type="text"
                    name="title"
                    className="form-control form-control-lg"
                    placeholder="Enter Recipe name"
                  />
                  { this.state.titleError &&
                  <span className="help-block error text-danger">
                    {this.state.titleError}
                  </span>
                    }
                </div>

                <div className="form-group">
                  <label htmlFor="ingredient">Ingredients</label>
                  <input
                    onChange={this.onChange}
                    type="text"
                    name="ingredients"
                    className="form-control form-control-lg"
                    placeholder="Enter Ingredients"
                    id="ingredients"
                  />
                  { this.state.ingredientError &&
                    <span className="help-block error text-danger">
                      { this.state.ingredientError}
                    </span>
                    }
                </div>

                <div className="form-group">
                  <label htmlFor="instruction">Instructions</label>
                  <textarea
                    onChange={this.onChange}
                    className="form-control form-control-lg"
                    placeholder="Enter Instructions"
                    name="instructions"
                    id="instructions"
                  />
                  { this.state.instructionError &&
                  <span className="help-block error text-danger">
                    { this.state.instructionError}
                  </span>
                    }
                </div>
                <Loader
                  loaded={this.state.loaded}
                  lines={13}
                  length={30}
                  width={10}
                  radius={30}
                  corners={1}
                  rotate={0}
                  direction={1}
                  color="#000"
                  speed={2}
                  trail={60}
                  shadow={false}
                  hwaccel={false}
                  className="spinner"
                  zIndex={2e9}
                  top="70%"
                  left="50%"
                  scale={1.00}
                  loadedClassName="loadedContent"
                />

                <div className="form-group">
                  <input
                    type="submit"
                    className="btn btn-success"
                    disabled={this.state.addingRecipe}
                    value={this.state.addingRecipe ? 'Adding Recipe...' :
                     'Add Recipe'}
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


AddRecipeForm.defaultProps = {
  addRecipe: {}
};

AddRecipeForm.propTypes = {
  addRecipe: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired
};


const mapStateToProps = state => ({
  recipes: state.recipes,
  loaded: state.loaded
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ addRecipe }, dispatch);

const ConnectedAddRecipeForm =
  connect(mapStateToProps, mapDispatchToProps)(AddRecipeForm);

export default ConnectedAddRecipeForm;
