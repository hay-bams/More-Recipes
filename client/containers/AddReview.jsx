import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Authenticate from '../auth/auth';
import { addReview } from '../actions/actions';

/**
 * @class AddReview
 */
class AddReview extends React.Component {
  /**
   * @param {obj} event
   * @returns {void} clearForm
   */
  static clearForm(event) {
    event.target.review.value = '';
  }

  /**
   * @returns {void} constructor
   */
  constructor() {
    super();
    this.addReview = this.addReview.bind(this);
    this.state = {
      errors: {}
    };
  }
  /**
   * @param {obj} event
   * @returns {void} addReview
   */
  addReview(event) {
    event.preventDefault();
    const userReview = {
      review: event.target.review.value
    };

    const errors = Authenticate.validateReview(userReview);

    if (errors.review !== '') {
      return this.setState({ errors });
    }

    const recipeId = parseInt(this.props.match.params.id, 10);
    this.props.addReview(userReview, recipeId);
    AddReview.clearForm(event);
  }

  /**
   * @returns {void} render
   */
  render() {
    const { errors } = this.state;
    return (
      <div className="container">
        <div className="row review-form">
          <div className="col-lg-6">
            <h4>Add Review</h4>
            <form onSubmit={this.addReview}>
              <div className="form-group">
                <textarea className="form-control" name="review" />
                { errors.review &&
                  <span className="help-block error text-danger">
                    {errors.review}
                  </span>
                    }
              </div>

              <div className="form-group">
                <input type="submit" className="btn btn-success" />
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

AddReview.defaultProps = {
  addReview: {},
};

AddReview.propTypes = {
  addReview: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired
};

const mapStateToProps = state => ({
  UserReviews: state.reviews
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ addReview }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AddReview);

