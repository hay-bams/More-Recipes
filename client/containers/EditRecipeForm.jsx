import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { editRecipe } from '../actions/actions';
import Authenticate from '../auth/auth';

/**
 * @class EditRecipeForm
 */
class EditRecipeForm extends React.Component {
  /**
   * @param {obj} props
   * @returns {void} constructor
   */
  constructor(props) {
    super(props);
    this.editRecipe = this.editRecipe.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      recipe: {},
      errors: {}
    };
  }

  /**
   * @returns {void} componentWillMount
   */
  componentWillMount() {
    this.props.recipes.filter((recipe) => {
      if (recipe.id === parseInt(this.props.match.params.id, 10)) {
        this.setState({ recipe });
      }
    });
  }

  /**
   * @param {obj} event
   * @returns {void} onChange
   */
  onChange(event) {
    const recipe = {
      [event.target.name]: event.target.value
    };
    this.setState({ recipe });
  }

  /**
   * @param {obj} event
   * @returns {void} addRecipe
   */
  editRecipe(event) {
    event.preventDefault();
    const recipe = {
      title: event.target.title.value,
      ingredients: event.target.ingredients.value,
      instructions: event.target.instructions.value,
      image: 'imageURl'
    };

    let errors = Authenticate.validateAddRecipe(recipe);

    if (errors.title !== '' || errors.ingredients !== '' ||
     errors.instructions) {
      return this.setState({ errors });
    }

    this.props.editRecipe(recipe, this.props.match.params.id);
    // AddRecipeForm.clearForm(event);

    errors = {};
    this.setState({ errors });
  }

  /**
   * @returns {obj} render
   */
  render() {
    const { errors } = this.state;
    const { recipe } = this.state;
    return (
      <div className="main-userboard-body add-recipe-body">
        <div className="container">
          <div className="row">
            <div className="mx-auto col-sm-12 col-md-8 col-lg-6 col-xs mt-5">
              <h2 className="text-center"> Edit Recipe</h2>
              <form onSubmit={this.editRecipe}>
                <div className="form-group">
                  <label htmlFor="food">Recipe Name</label>
                  <input
                    type="text"
                    name="title"
                    className="form-control"
                    placeholder="Enter Recipe name"
                    value={recipe.title}
                    onChange={this.onChange}
                  />
                  { errors.title &&
                  <span className="help-block text-danger error">
                    {errors.title}
                  </span>
                    }
                </div>

                <div className="form-group">
                  <input type="file" />
                </div>

                <div className="form-group">
                  <label htmlFor="ingredient">Ingredients</label>
                  <input
                    type="text"
                    name="ingredients"
                    className="form-control"
                    placeholder="Enter Ingredients"
                    value={recipe.ingredients}
                    onChange={this.onChange}
                  />
                  { errors.ingredients &&
                    <span className="help-block text-danger error">
                      {errors.ingredients}
                    </span>
                    }
                </div>

                <div className="form-group">
                  <label htmlFor="instruction">Instructions</label>
                  <textarea
                    className="form-control"
                    placeholder="Enter Instructions"
                    name="instructions"
                    id="instructions"
                    value={recipe.instructions}
                    onChange={this.onChange}
                  />
                  { errors.instructions &&
                  <span className="help-block text-danger error">
                    {errors.instructions}
                  </span>
                    }
                </div>

                <div className="form-group">
                  <input
                    type="submit"
                    className="btn btn-success"
                    value="Update Recipe"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>

    );
  }
}


EditRecipeForm.defaultProps = {
  editRecipe: {}
};

EditRecipeForm.propTypes = {
  editRecipe: PropTypes.func,
  recipes: PropTypes.arrayOf(PropTypes.shape({
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
    instructions: PropTypes.string,
    ingredients: PropTypes.string,
    userId: PropTypes.number,
    createdAt: PropTypes.string,
    updatedAt: PropTypes.string
  })).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired
};


const mapStateToProps = state => ({
  recipes: state.userRecipes
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ editRecipe }, dispatch);


export default connect(mapStateToProps, mapDispatchToProps)(EditRecipeForm);
