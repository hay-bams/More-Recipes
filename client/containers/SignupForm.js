import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'; 
import { signup } from '../actions/actions.js';
import Authenticate from '../auth/auth.js';

/**
 * @class SignupForm
 */
class SignupForm extends React.Component {
  constructor(props) {
    super(props);
    this.message = "";
  }

  componentDidUpdate() {
    this.props.user.userData !== undefined ? this.props.redirectUser.push('/dashboard') : "";
  }

  signup(e) {
    e.preventDefault();
    const user = {
      firstName: this.firstName.value,
      lastName: this.lastName.value,
      email: this.email.value,
      password: this.password.value,
      confirmPassword: this.confirmPassword.value
    }
    this.message = Authenticate.validateUserSignup(user);
    this.props.signup(user);
    this.message = this.message ? this.message : "Email already exist";
  }

  render() {
      return(
          <div className = "container-fluid main-register-container">
          <div className = "row">
            <div className = "col-sm-8 col-md-5 mx-auto">
              <div className = "card  mt-5">
                <div className = "card-header">
                  Register 
                </div>
    
                <div className = "card-body">
                <span>{this.message}</span>
                  <form onSubmit={(e) => this.signup(e)}>
                      <div className = "form-group">
                      <label htmlFor = "first_name">First Name</label>
                      <input ref={(input) => this.firstName = input} type="text" name="firstName" className = "form-control" placeholder="Enter First name" />
                    </div>
    
                      <div className = "form-group">
                      <label htmlFor = "last_name">Last Name</label>
                      <input ref={(input) => this.lastName = input} type="text" name="lastName" className = "form-control" placeholder="Enter Last name" />
                    </div>
    
                    <div className = "form-group">
                      <label htmlFor = "email">Email</label>
                      <input ref={(input) => this.email = input} type="email" name="email" className = "form-control" placeholder="Enter Email" />
                    </div>
    
                    <div className = "form-group">
                      <label htmlFor = "password">Password</label>
                      <input ref={(input) => this.password = input} type="password" name="password" className = "form-control" placeholder="Enter Password" />
                    </div>
    
                      <div className = "form-group">
                      <label htmlFor = "password">Confirm Password</label>
                      <input ref={(input) => this.confirmPassword = input} type="password" name="Confirmpassword" className = "form-control" placeholder="Confirm Password" />
                    </div>
    
                    <div className = "form-group">
                      <input type="submit" value = "Register" className = "btn btn-success btn-block" />
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      )
  }
}

const mapStateToProps = state => {
  return {
    user: state.user
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({signup}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(SignupForm);