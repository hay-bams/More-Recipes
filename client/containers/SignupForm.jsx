import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { signup } from '../actions/user';
import Authenticate from '../auth/auth';

/**
 * @class SignupForm
 */
export class SignupForm extends React.Component {
  /**
   * @returns{void} constructor
   */
  constructor() {
    super();
    this.message = '';
    this.signup = this.signup.bind(this);
    this.onChange = this.onChange.bind(this);
    this.state = {
      firstName: null,
      lastName: null,
      email: '',
      password: null,
      confirmPassword: null,
      firstNameError: '',
      lastNameError: '',
      emailError: '',
      passwordError: '',
      confirmPasswordError: ''
    };
  }

  /**
   * @returns {void} componentDidUpdate
   */
  componentDidUpdate() {
    if (Object.keys(this.props.userData).length > 0) {
      this.props.redirectUser.push('/');
    }
  }

  /**
  *
  * @param {obj} event
  * @returns {void} onChange
  */
  onChange(event) {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @param {obj} event
   * @returns {void} signup
   */
  signup(event) {
    event.preventDefault();
    const user = {
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      email: this.state.email,
      password: this.state.password,
      confirmPassword: this.state.confirmPassword
    };
    const errors = Authenticate.validateUserSignup(user);

    if (errors.email !== '' || errors.password !== '' ||
       errors.firstName !== '' || errors.lastName !== '' ||
       errors.confirmPassword !== '') {
      return this.setState({
        firstNameError: errors.firstName,
        lastNameError: errors.lastName,
        emailError: errors.email,
        passwordError: errors.password,
        confirmPasswordError: errors.confirmPassword
      });
    }

    this.props.signup(user);
  }

  /**
   * @returns {void} render
   */
  render() {
    return (
      <div className="container-fluid main-register-container">
        <div className="row overlay">
          <div className="col-sm-8 col-md-5 mx-auto">
            <div className="card  mt-5 card-form">
              <div className="card-header text-center">
               Sign Up
              </div>
              <div className="card-body">
                <form onSubmit={e => this.signup(e)}>
                  <div className="form-group">
                    <input
                      onChange={this.onChange}
                      type="text"
                      name="firstName"
                      className="form-control"
                      placeholder="Enter First name"
                    />
                    { this.state.firstNameError &&
                    <span className="help-block error text-danger">
                      {this.state.firstNameError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      onChange={this.onChange}
                      type="text"
                      name="lastName"
                      className="form-control"
                      placeholder="Enter Last name"
                    />
                    { this.state.lastNameError &&
                    <span className="help-block error text-danger">
                      {this.state.lastNameError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      onChange={this.onChange}
                      type="email"
                      name="email"
                      className="form-control"
                      placeholder="Enter Email"
                    />
                    { this.state.emailError &&
                    <span className="help-block error text-danger">
                      {this.state.emailError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      onChange={this.onChange}
                      type="password"
                      name="password"
                      className="form-control"
                      placeholder="Enter Password"
                    />
                    { this.state.passwordError &&
                    <span className="help-block error text-danger">
                      {this.state.passwordError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      onChange={this.onChange}
                      type="password"
                      name="confirmPassword"
                      className="form-control"
                      placeholder="Confirm Password"
                    />
                    { this.state.confirmPasswordError &&
                    <span className="help-block error text-danger">
                      {this.state.confirmPasswordError}
                    </span>
                      }
                  </div>

                  <div className="form-group">
                    <input
                      type="submit"
                      value="Register"
                      className="btn btn-success btn-block"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  userData: state.userData
});

SignupForm.defaultProps = {
  userData: {},
  redirectUser: {}
};

SignupForm.propTypes = {
  userData: PropTypes.shape({
    user: PropTypes.shape({
      email: PropTypes.string,
      firstName: PropTypes.string,
      id: PropTypes.number,
      lastName: PropTypes.string
    })
  }),
  signup: PropTypes.func.isRequired,
  redirectUser: PropTypes.shape({
    push: PropTypes.func
  })
};

const mapDispatchToProps = dispatch => bindActionCreators({ signup }, dispatch);

const ConnectedSignupForm =
connect(mapStateToProps, mapDispatchToProps)(SignupForm);

export default ConnectedSignupForm;

