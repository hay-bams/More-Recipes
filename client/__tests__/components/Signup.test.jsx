import React from 'react';
import { shallow } from 'enzyme';
import { SignupForm } from '../../containers/SignupForm';


describe('Signin component', () => {
  const props = {
    signup: jest.fn(),
    userData: {
      user: {
        email: 'test@gmail.com',
        firstName: 'Ayobami',
        id: 1,
        lastName: 'Adelakun'
      }
    },
    redirectUser: {
      push: jest.fn()
    }
  };

 
  test('it should render SignupForm correctly', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    expect(wrapper.find('form').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('it should render error for an empty email in form submission', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.instance().setState({
      email: '',
      password: 'password',
      firstName: 'Ayobami',
      lastName: 'Adelakun',
      confirmPassword: 'password'
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state('emailError').length).toBeGreaterThan(0);
  });

  test('it should render error for empty password in form submission', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.instance().setState({
      email: 'email@gmail.com',
      password: '',
      firstName: 'Ayobami',
      lastName: 'Adelakun',
      confirmPassword: 'password'
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state('passwordError').length).toBeGreaterThan(0);
  });


  test('it should render error for empty firstName in form submission', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.instance().setState({
      email: 'email@gmail.com',
      password: 'password',
      firstName: '',
      lastName: 'Adelakun',
      confirmPassword: 'password'
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state('firstNameError').length).toBeGreaterThan(0);
  });

  test('it should render error for empty firstName in form submission', () => {
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.instance().setState({
      email: 'email@gmail.com',
      password: 'password',
      firstName: 'Ayobami',
      lastName: '',
      confirmPassword: 'password'
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state('lastNameError').length).toBeGreaterThan(0);
  });

  test('should set firstName state on input change', () => {
    const value = 'Ayobami';
    const name = 'firstName';
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
      preventDefault: () => {}
    });

    expect(wrapper.state('firstName')).toBe(value);
  });

  test('should set lastName state on input change', () => {
    const value = 'Adelakun';
    const name = 'lastName';
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
      preventDefault: () => {}
    });

    expect(wrapper.state('lastName')).toBe(value);
  });

  test('should set email state on input change', () => {
    const value = 'email@gmail.com';
    const name = 'email';
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
      preventDefault: () => {}
    });

    expect(wrapper.state('email')).toBe(value);
  });

  test('should set password state on input change', () => {
    const name = 'password';
    const value = 'password';
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
      preventDefault: () => {}
    });

    expect(wrapper.state('password')).toBe(value);
  });

  test('should set confirmPassword state on input change', () => {
    const name = 'confirmPassword';
    const value = 'password';
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
      preventDefault: () => {}
    });

    expect(wrapper.state('confirmPassword')).toBe(value);
  });

  test('should call signup prop for valid form submission', () => {
    const signupInfo = {
      email: 'email@gmail.com',
      password: 'password',
      firstName: 'Ayobami',
      lastName: 'Adelakun',
      confirmPassword: 'password'
    };
    const wrapper = shallow(<SignupForm {...props} />);
    wrapper.instance().setState({
      email: 'email@gmail.com',
      password: 'password',
      firstName: 'Ayobami',
      lastName: 'Adelakun',
      confirmPassword: 'password'
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });

    expect(props.signup).toHaveBeenLastCalledWith(signupInfo);
  });
});

