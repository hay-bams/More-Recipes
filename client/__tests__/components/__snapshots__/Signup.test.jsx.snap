// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin component it should render SignupForm correctly 1`] = `
<div
  className="container-fluid main-register-container"
>
  <div
    className="row overlay"
  >
    <div
      className="col-sm-8 col-md-5 mx-auto"
    >
      <div
        className="card  mt-5 card-form"
      >
        <div
          className="card-header text-center"
        >
          Sign Up
        </div>
        <div
          className="card-body"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="firstName"
                onChange={[Function]}
                placeholder="Enter First name"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="lastName"
                onChange={[Function]}
                placeholder="Enter Last name"
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Enter Email"
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Enter Password"
                type="password"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="btn btn-success btn-block"
                id="registerbutton"
                type="submit"
                value="Register"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
